benchmarks/benchmark-solve-cra.C

```diff
@@ -49,18 +49,19 @@ using Field = Givaro::ZRing<Integer>;
 #endif
 
 template <class Field, class Matrix>
-static bool checkResult(const Field& ZZ, Matrix& A, BlasVector<Field>& B, BlasVector<Field>& X, Integer& d)
+static bool checkResult(Matrix& A, BlasVector<Field>& B, BlasVector<Field>& X, Integer& d)
 {
+    Field ZZ;
     BlasVector<Field> B2(ZZ, A.coldim());
     BlasVector<Field> B3(ZZ, A.coldim());
     A.apply(B2, X);
-
-    Integer tmp;
+   
     for (size_t j = 0; j < B.size(); ++j) {
-        B3.setEntry(j, d * B.getEntry(j));
+        ZZ.mul(B3[j],B[j],d);
```

Comment:
Why don't you use a vector domain, as I suggested? (avoiding to write the for loop)

Reply:
d here is just a constant value not a vector, I wonder if it would be better to declare a vector to use the suggested function instead of the current implementation.